prow/kube/prowjob.go

```diff
@@ -79,22 +79,45 @@ type ProwJob struct {
 }
 
 type ProwJobSpec struct {
-	Type    ProwJobType  `json:"type,omitempty"`
-	Agent   ProwJobAgent `json:"agent,omitempty"`
-	Cluster string       `json:"cluster,omitempty"`
-	Job     string       `json:"job,omitempty"`
-	Refs    *Refs        `json:"refs,omitempty"`
-
+	Type      ProwJobType  `json:"type,omitempty"`
+	Agent     ProwJobAgent `json:"agent,omitempty"`
+	Cluster   string       `json:"cluster,omitempty"`
+	Job       string       `json:"job,omitempty"`
+	Refs      *Refs        `json:"refs,omitempty"`
+	ExtraRefs []*Refs      `json:"extra_refs,omitempty"`
+
+	TimeoutMinutes int    `json:"timeout_minutes,omitempty"`
 	Report         bool   `json:"report,omitempty"`
 	Context        string `json:"context,omitempty"`
 	RerunCommand   string `json:"rerun_command,omitempty"`
 	MaxConcurrency int    `json:"max_concurrency,omitempty"`
 
+	Decorate         bool             `json:"decorate,omitempty"`
+	UtilityImages    UtilityImages    `json:"utility_images,omitempty"`
+	GCSConfiguration GCSConfiguration `json:"gcs_configuration,omitempty"`
+
 	PodSpec *v1.PodSpec `json:"pod_spec,omitempty"`
 
 	RunAfterSuccess []ProwJobSpec `json:"run_after_success,omitempty"`
 }
 
+// UtilityImages holds pull specs for the utility images
+// to be used for a job
+type UtilityImages struct {
+	CloneRefs  string `json:"clonerefs,omitempty"`
+	InitUpload string `json:"initupload,omitempty"`
+	Entrypoint string `json:"entrypoint,omitempty"`
+	Sidecar    string `json:"sidecar,omitempty"`
+}
+
+type GCSConfiguration struct {
+	Bucket            string `json:"bucket,omitempty"`
+	PathStrategy      string `json:"path_strategy,omitempty"`
+	DefaultOrg        string `json:"default_org,omitempty"`
+	DefaultRepo       string `json:"default_repo,omitempty"`
+	CredentialsSecret string `json:"credentials_secret,omitempty"`
```

Comment:
Is this a path or a secret ref? All of these fields need godoc

Reply:
secret name, can doc